packages/filesystem/src/browser/file-tree/file-tree-model.ts

```diff
@@ -12,6 +12,8 @@ import { FileSystem, FileSystemWatcher, FileChangeType, FileChange } from "../..
 import { FileStatNode, DirNode, FileTree } from "./file-tree";
 import { LocationService } from '../location';
 import { LabelProvider } from "@theia/core/lib/browser/label-provider";
+import * as base64 from 'base64-js';
```

Comment:
I am not sure that we need this additional dependency. There seems to be already APIs to handle base64 encoding in the browser env: https://stackoverflow.com/questions/6978156/get-base64-encode-file-data-from-input-form

This package should be used in pure js.

Reply:
The alternative is to use readAsDataURL, which creates a data URL of the form

    data:application/x-desktop;base64,<base64 data>

But doing some string processing to extract the base64 data from the URL seems more hacky and error-prone than using a package (which is really small) that does ArrayBuffer -> base64 directly.  The exact format of the URL returned by readAsDataURL is probably not guaranteed to be always exactly the same across browsers, browser version, etc.